grove/bernstein_vazirani/bernstein_vazirani.py

```diff
@@ -179,17 +178,30 @@ def run(self, cxn, bitstring_map):
         self.bv_circuit = self._create_bv_circuit(bitstring_map)
 
         # find vector by running the full bv circuit
-        results = cxn.run_and_measure(self.bv_circuit, self.computational_qubits)
-        bv_vector = results[0][::-1]
+        full_circuit = Program()
+        full_ro = full_circuit.declare('ro', 'BIT', len(self.computational_qubits) + 1)
+        full_circuit += self.bv_circuit
+        full_circuit += [MEASURE(qubit, ro) for qubit, ro in zip(self.computational_qubits,
+                                                                 full_ro)]
+        full_circuit.wrap_in_numshots_loop(1)
```

Comment:
You don't need this line at all if you're only doing a single shot. You may have to change the indexing in `full_results` to obtain `bv_vector` below though if you do remove this line.

Reply:
Correcting myself, you probably *don't* need to change the indexing of `full_results` below, if you remove this line, since `Program().wrap_in_numshots_loop(1)` seems to happen by default anyway.